main/templates/main/messages.html

```diff
@@ -13,13 +13,31 @@ <h5 class="d-inline">Messages</h5>
     <div class="card-body">
       {% now "e (O)" as server_time %}
       <span id="timezone" class="d-line text-body-tertiary">All timestamps are displayed in {{ server_time }}</span>
-      <input class="form-control mb-2"
-             type="search"
-             name="search"
-             placeholder="Search messages..."
-             hx-get="{% url 'main:messages' topic|default:"ALL" %}"
-             hx-trigger="input changed delay:500ms, every 1s"
-             hx-target="#message-list">
+      <!-- Filter Form -->
+      <form id="filter-form" method="get">
+        <!-- Search Input -->
+        <input class="form-control mb-2"
+               type="search"
+               name="search"
+               placeholder="Search messages..."
+               hx-get="{% url 'main:messages' topic|default:"ALL" %}"
+               hx-trigger="input changed delay:500ms, every 1s"
+               hx-target="#message-list"
+               hx-include="#filter-form">
+        <!-- Severity Filter Dropdown -->
+        <select name="severity"
+                class="form-select mb-2"
+                hx-get="{% url 'main:messages' topic|default:"ALL" %}"
+                hx-trigger="change delay:500ms"
```

Comment:
don't think we need the delay here

Reply:
Yeah, I did try to play around with adding multiple but would need to add some custom JavaScript for that

Reply:
I tried out Django filter but was having issues with the constant refreshing of the message pane. I could have found a workaround with custom JavaScript, but I thought this was cleaner.